frontend/javascripts/oxalis/model/sagas/load_histogram_data_saga.ts

```diff
@@ -1,88 +1,99 @@
-import type { HistogramDataForAllLayers } from "oxalis/store";
+import type { HistogramDataForAllLayers, DatasetLayerConfiguration } from "oxalis/store";
 import type { Saga } from "oxalis/model/sagas/effect-generators";
 import { select } from "oxalis/model/sagas/effect-generators";
-import { call, take, put } from "typed-redux-saga";
+import { call, take, takeEvery, put } from "typed-redux-saga";
 import {
   setHistogramDataAction,
+  setHistogramDataForLayerAction,
   updateLayerSettingAction,
+  type ReloadHistogramAction,
 } from "oxalis/model/actions/settings_actions";
 import { getHistogramForLayer } from "admin/admin_rest_api";
 import DataLayer from "oxalis/model/data_layer";
 import Model from "oxalis/model";
-import type { APIDataset } from "types/api_flow_types";
+import type { APIHistogramData } from "types/api_flow_types";
 
-async function fetchAllHistogramsForLayers(
-  dataLayers: Array<DataLayer>,
-  dataset: APIDataset,
-): Promise<HistogramDataForAllLayers> {
-  const histograms: HistogramDataForAllLayers = {};
+export default function* loadHistogramDataSaga(): Saga<void> {
+  yield* take("WK_READY");
+  yield* takeEvery("RELOAD_HISTOGRAM", reloadHistogramForLayer);
+  const dataLayers: Array<DataLayer> = yield* call([Model, Model.getColorLayers]) as any;
+  const layerConfigurations = yield* select((state) => state.datasetConfiguration.layers);
 
+  const histograms: HistogramDataForAllLayers = {};
   for (const dataLayer of dataLayers) {
-    try {
-      // We send the histogram data requests sequentially so there is less blockage of bucket data requests.
-      // eslint-disable-next-line no-await-in-loop
-      const data = await getHistogramForLayer(dataset.dataStore.url, dataset, dataLayer.name);
-
-      if (Array.isArray(data) && data.length > 0) {
-        histograms[dataLayer.name] = data;
-      }
-    } catch (e) {
-      console.warn(`Error: Could not fetch the histogram data for layer ${dataLayer.name}.`, e);
-    }
+    const layerName = dataLayer.name;
+    // We send the histogram data requests sequentially so there is less blockage of bucket data requests.
+    const histogram = yield* call(loadHistogramForLayer, layerName, layerConfigurations[layerName]);
```

Comment:
I think, it would simplify things if `loadHistogramForLayer` doesn't take the `layerConfiguration` parameter (especially, because there are two call sites where this needs to be prepared).